compiler/rustc_lint/src/if_let_rescope.rs

```diff
@@ -145,7 +149,10 @@ impl IfLetRescope {
                 recovered: Recovered::No,
             }) = cond.kind
             {
-                let if_let_pat = expr.span.shrink_to_lo().between(init.span);
+                // Peel off round braces
+                let if_let_pat = source_map
+                    .span_take_while(expr.span, |&ch| ch == '(' || ch.is_whitespace())
```

Comment:
Question: does this handle weird things like `if ((((if let .. { .. } else { .. }))))` correctly? Even though I don't think this is very common...